### ISSUE TITLE
numpy 2+

### ISSUE THREAD

**Issue by SanPen** [ramonhagenaars/nptyping#125]
Hi, 

I use nptyping, and I needed it to work with numpy 2.x, hence I just copied the code in my repo and referenced it.

The types from the file `typing_.py` that work for both new and old numpy are: 

```python
try:
    from typing import (  # type: ignore[attr-defined,misc] # pylint: disable=unused-import
        Literal,
        TypeAlias,
        TypeGuard,
        final,
    )
except ImportError:  # pragma: no cover
    from typing_extensions import (  # type: ignore[attr-defined,misc]
        Literal,
        TypeAlias,
        TypeGuard,
        final,
    )

from typing import Tuple, Union

import numpy as np

ShapeExpression: TypeAlias = str
StructureExpression: TypeAlias = str
DType: TypeAlias = Union[np.generic, StructureExpression]
ShapeTuple: TypeAlias = Tuple[int, ...]

Number = np.number
Bool = np.bool_
Bool8 = np.bool_
Obj = np.object_  # Obj is a common abbreviation and should be usable.
Object = np.object_
Object0 = np.object_
Datetime64 = np.datetime64
Integer = np.integer
SignedInteger = np.signedinteger
Int8 = np.int8
Int16 = np.int16
Int32 = np.int32
Int64 = np.int64
Byte = np.byte
Short = np.short
IntC = np.intc
IntP = np.intp
Int0 = np.int8
Int = np.integer  # Int should translate to the "generic" int type.
Int_ = np.int_
LongLong = np.longlong
Timedelta64 = np.timedelta64
UnsignedInteger = np.unsignedinteger
UInt8 = np.uint8
UInt16 = np.uint16
UInt32 = np.uint32
UInt64 = np.uint64
UByte = np.ubyte
UShort = np.ushort
UIntC = np.uintc
UIntP = np.uintp
UInt0 = np.uint
UInt = np.uint
ULongLong = np.ulonglong
Inexact = np.inexact
Floating = np.floating
Float16 = np.float16
Float32 = np.float32
Float64 = np.float64
Half = np.half
Single = np.single
Double = np.double
Float = np.float64
LongDouble = np.longdouble
LongFloat = np.longdouble
ComplexFloating = np.complexfloating
Complex64 = np.complex64
Complex128 = np.complex128
CSingle = np.csingle
SingleComplex = np.complex64
CDouble = np.cdouble
Complex = np.complex128
CFloat = np.complex128
CLongDouble = np.clongdouble
CLongFloat = np.clongdouble
LongComplex = np.clongdouble
Flexible = np.flexible
Void = np.void
Void0 = np.void
Character = np.character
Bytes = np.bytes_
Str = np.str_
String = np.bytes_
Bytes0 = np.bytes_
Unicode = np.str_
Str0 = np.str_

dtypes = [
    (Number, "Number"),
    (Bool, "Bool"),
    (Bool8, "Bool8"),
    (Obj, "Obj"),
    (Object, "Object"),
    (Object0, "Object0"),
    (Datetime64, "Datetime64"),
    (Integer, "Integer"),
    (SignedInteger, "SignedInteger"),
    (Int8, "Int8"),
    (Int16, "Int16"),
    (Int32, "Int32"),
    (Int64, "Int64"),
    (Byte, "Byte"),
    (Short, "Short"),
    (IntC, "IntC"),
    (IntP, "IntP"),
    (Int0, "Int0"),
    (Int, "Int"),
    (LongLong, "LongLong"),
    (Timedelta64, "Timedelta64"),
    (UnsignedInteger, "UnsignedInteger"),
    (UInt8, "UInt8"),
    (UInt16, "UInt16"),
    (UInt32, "UInt32"),
    (UInt64, "UInt64"),
    (UByte, "UByte"),
    (UShort, "UShort"),
    (UIntC, "UIntC"),
    (UIntP, "UIntP"),
    (UInt0, "UInt0"),
    (UInt, "UInt"),
    (ULongLong, "ULongLong"),
    (Inexact, "Inexact"),
    (Floating, "Floating"),
    (Float16, "Float16"),
    (Float32, "Float32"),
    (Float64, "Float64"),
    (Half, "Half"),
    (Single, "Single"),
    (Double, "Double"),
    (Float, "Float"),
    (LongDouble, "LongDouble"),
    (LongFloat, "LongFloat"),
    (ComplexFloating, "ComplexFloating"),
    (Complex64, "Complex64"),
    (Complex128, "Complex128"),
    (CSingle, "CSingle"),
    (SingleComplex, "SingleComplex"),
    (CDouble, "CDouble"),
    (Complex, "Complex"),
    (CFloat, "CFloat"),
    (CLongDouble, "CLongDouble"),
    (CLongFloat, "CLongFloat"),
    (LongComplex, "LongComplex"),
    (Flexible, "Flexible"),
    (Void, "Void"),
    (Void0, "Void0"),
    (Character, "Character"),
    (Bytes, "Bytes"),
    (String, "String"),
    (Str, "Str"),
    (Bytes0, "Bytes0"),
    (Unicode, "Unicode"),
    (Str0, "Str0"),
]

name_per_dtype = dict(dtypes)
dtype_per_name = {name: dtype for dtype, name in dtypes}
```

Would be cool if you fix it and push, otherwise here it is for others to use.

**Comment by maxsegl** [comment]
Would make my life also easier!

**Comment by fherb2** [comment]
The same as maxsegl for my life.
:-)

**Comment by shaunc** [comment]
@fherb2 - thank you for the fork, though! You say "private" but for the time being I have referenced!

https://github.com/fherb2/nptyping

👍

**Comment by burnpanck** [comment]
It looks like @ramonhagenaars currently isn't able or willing to work on this project. Maybe the community could agree on a new home to continue their work - even if that home is just temporary. Apart from @fherb2's fork, there is at least @jasper-tms contending with [their fork](https://github.com/jasper-tms/nptyping), as suggested in a [comment](https://github.com/ramonhagenaars/nptyping/pull/114#issuecomment-2200563298) to PR #114, which is what I have been using successfully in the past. Anyone of you willing to pitch in here and willing to maintain a temporary home for this project?

**Comment by jasper-tms** [comment]
I don't have interest in working on the library myself, but I certainly plan to be doing computer things and being responsive to messages on github for the next... decades. If someone else wants to do the work, I could certainly give the repo a home. I would imagine it makes more sense for someone who has more concrete interest in the package than me to provide the home though.

**Comment by jasper-tms** [comment]
Update: My fork exists https://github.com/jasper-tms/nptyping, it has the basic changes needed for compatibility with numpy 2.x, and I'll be happy to merge any PRs that people submit. Thanks @shaunc for having contributed one.

I suggest we leave this issue open for visibility. @SanPen, if you don't mind, could you update the title of this issue to start with `[Solved]` or something along those lines, and perhaps edit your initial comment to put

> tldr: Use https://github.com/jasper-tms/nptyping

near the top? Thanks all.

If people want, I can push the updated package to pypi as `np2typing`/`nptyping2` or something for easier installation via `pip install np2typing`. Opinions?

**Comment by feitel-db** [comment]
> 
> If people want, I can push the updated package to pypi as `np2typing`/`nptyping2` or something for easier installation via `pip install np2typing`. Opinions?

Yes a push to pypi would be great! It seems nptyping2 is already in use though..

**Comment by jasper-tms** [comment]
OK done: https://pypi.org/project/np2typing

Some notes:
1. The package is still named `nptyping` so `import nptyping` is still the correct import command, but it's on pypi as `np2typing`. So if people just switch out `nptyping` for `np2typing` in their requirements list, all should work.
2. I added python versions 3.12, 3.13, and 3.14 to the list of supported versions, but I did not test that this actually works – I wanted pip to allow people to proceed with installing the package on more recent python versions, but I haven't checked if the library actually runs correctly on these python versions. If anyone wants to help out and see if it work, feel free.
3. I was able to update some of the badges at the start of the README, but the downloads badge and the codecov badge currently still point to the original `nptyping`, because it looks like I would have to configure some things to get them to work with `np2typing`. If anyone wants to look into setting that up, the help would be happily taken.
